pypsa/optimization/optimize.py

```diff
@@ -46,14 +51,18 @@
     define_transmission_volume_expansion_limit,
 )
 from pypsa.optimization.variables import (
+    # Removed function no longer used
+    # define_start_up_variables,
+    # Removed function no longer used
+    # define_status_variables,
+    define_integer_committability_variables,
```

Comment:
I'd recommend to avoid changing too much the structure; better to keep the previous functions and adapt them.
What do you think?

Reply:
It works for me. The idea behind "one single function" follows.

PyPSA now has 3 functions to create the variables status, startup and shutdown. Since these variables are always binary, we only have 3 functions that do the same thing by assigning different names to the variables (each function creates the variable and also sets the constraint <=1).

Going back to our case, for each of the 3 variables mentioned, we need
a) a function to create the integer variable
b) a function to limit the variable to <= n_mod variable (when referring to expandable components)
c) a function to limit the variable to <= P_nom/P_nom_mod (when referring to a non-extendable component).
d) a function to limit the variable to <= 1 (when modularity is not used - see actual situation).

It follows that in our case the 3 functions would become 12. To avoid this, and to limit the "increase of functions" (that you mention in the previous comment), I prefer to use only one function for each of the 4 actions a), b), c) ,d). 

However, having clarified the idea behind it, if you suggest keeping 9 different functions, I'll move in that direction.